clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/AbstractKafkaConfigProperties.java

```diff
@@ -50,15 +51,15 @@ public abstract class AbstractKafkaConfigProperties {
      * Creates a new instance.
      */
     protected AbstractKafkaConfigProperties() {
-        this.componentUId = getComponentUIdFromEnv();
+        this.componentUId = COMPONENT_UID_DEFAULT;
     }
 
-    private String getComponentUIdFromEnv() {
-        final String k8sContainerId = KubernetesContainerUtil.getContainerId();
-        if (k8sContainerId != null && k8sContainerId.length() >= 12) {
+    private static String getK8sComponentUId() {
+        if (System.getenv("KUBERNETES_SERVICE_HOST") != null) {
             // running in Kubernetes: use HOSTNAME env var containing the pod name
             final String podName = System.getenv("HOSTNAME");
-            return String.format("%s_%s", podName, k8sContainerId.substring(0, 12));
+            final String random = UUID.randomUUID().toString().replaceAll("-", "").substring(0, 12);
```

Comment:
is there a particular reason why you want to cap the UUID at 12 characters?
The UUID represents 128bit = 16 byte = 32 hex characters, so IMHO we are loosing a lot of the UUID's uniqueness probability here ...

Reply:
We already have the pod name in the ID. For the purpose of having distinct client IDs in logging/tracing (and in Kafka UIs showing the list of connected clients), I think this combination of pod name + 12char random ID is sufficiently distinct. The client id is already quite long - better to not make it longer.
In fact, I just adopted the 12char length from the previous code, I guess we could reduce it to 4 chars even.